azurerm/resource_arm_container_group.go

```diff
@@ -307,6 +307,154 @@ func resourceArmContainerGroup() *schema.Resource {
 								},
 							},
 						},
+
+						"liveness_probe": {
+							Type:     schema.TypeList,
+							Optional: true,
+							ForceNew: true,
+							MaxItems: 1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"exec": {
+										Type:     schema.TypeList,
+										Optional: true,
+										ForceNew: true,
+										Elem:     &schema.Schema{Type: schema.TypeString},
+									},
+
+									"httpget": {
+										Type:     schema.TypeList,
+										Optional: true,
+										ForceNew: true,
+										Elem: &schema.Resource{
+											Schema: map[string]*schema.Schema{
+												"path": {
+													Type:     schema.TypeString,
+													Optional: true,
+													ForceNew: true,
+												},
+												"port": {
+													Type:     schema.TypeInt,
+													Optional: true,
+													ForceNew: true,
+												},
+												"scheme": {
+													Type:     schema.TypeString,
+													Optional: true,
+													ForceNew: true,
+													ValidateFunc: validation.StringInSlice([]string{
+														string(containerinstance.HTTP),
+														string(containerinstance.HTTPS),
+													}, true),
+												},
+											},
+										},
+									},
+
+									"inital_delay_seconds": {
+										Type:     schema.TypeInt,
+										Optional: true,
+										ForceNew: true,
+									},
+
+									"period_seconds": {
+										Type:     schema.TypeInt,
+										Optional: true,
+										ForceNew: true,
+									},
+
+									"failure_threashold": {
+										Type:     schema.TypeInt,
+										Optional: true,
+										ForceNew: true,
+									},
+
+									"sucess_threashold": {
+										Type:     schema.TypeInt,
+										Optional: true,
+										ForceNew: true,
+									},
+
+									"timeout_seconds": {
+										Type:     schema.TypeInt,
+										Optional: true,
+										ForceNew: true,
+									},
+								},
+							},
+						},
+
+						"readiness_probe": {
```

Comment:
Given that these two properties share the exact same schema, would it make sense to create a helper function that returns the schema?

I am also not going to make the same comments above here.

Reply:
@katbyte yep, I was wondering how to properly handle this. I would be happy to do this. Are you aware of any other places in the provider that have a similar function that I can use as guidance?